### PR TITLE
refs function #11160 update-order-information

### PR DESCRIPTION
**Pull request by khanhtrant12**
Update order information
https://edu-redmine.sun-asterisk.vn/issues/11160

### CODE REVIEW

**Comment by khanhtrant12** [comment]
ready

File: app/controllers/order_details_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class OrderDetailsController < ApplicationController
 
   def create
     @order_detail = @order.order_details.new(order_detail_params)
-    @order.save
+    @order.save(validate: false)
```
Comment (huongnguyenmta) [inline app/controllers/order_details_controller.rb:6]: sao lại bỏ validate chỗ này đi em??

File: app/models/order.rb [hunk 2]
```diff
@@ -6,13 +6,19 @@ class Order < ApplicationRecord
...
+
   before_validation :update_order_status
   before_save :set_order_date, :update_subtotal
 
   def subtotal
     order_details.map{|od| od.valid? ? (od.quantity * od.unit_price) : 0}.sum
   end
 
+  def order_details_changed?
+    !order_details.empty?
```
Comment (huongnguyenmta) [inline app/models/order.rb]: ==> `order_details.present?`

**Comment by khanhtrant12** [comment]
ready

**Pull request merged**